extensions/amp-base-carousel/0.1/carousel-accessibility.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getDetail} from '../../../src/event-helper';
+
+/**
+ * Accessibility for the carousel. This behaves either as a list or by
+ * exposing a single item at a time using `aria-hidden`.
+ */
+export class CarouselAccessibility {
+  /**
+   * @param {{
+   *   win: !Window,
+   *   element: !Element,
+   *   scrollContainer: !Element,
+   *   runMutate: function(function()),
+   *   autoAdvance: !AutoAdvance,
+   * }} config
+   */
+  constructor({
+    win,
+    element,
+    scrollContainer,
+    runMutate,
+    autoAdvance,
+  }) {
+    /** @private @const */
+    this.win_ = win;
+
+    /** @private @const */
+    this.scrollContainer_ = scrollContainer;
+
+    /** @private @const */
+    this.runMutate_ = runMutate;
+
+    /** @private {!Array<!Element>} */
+    this.slides_ = [];
+
+    /** @private {number} */
+    this.visibleCount_ = 1;
+
+    /** @private {boolean} */
+    this.mixedLength_ = false;
+
+    element.addEventListener('focus', () => {
```

Comment:
To summarize our conversation offline, the action item here would be to clarify that these are internal-only functions (not exposed to the developer). If you ever decide that you want to expose these as developer oriented actions, it might be confusing to developers if we expose "pause/resume", but then that's permanently overridden by "stop", which is called on focus.

Reply:
Added comments to clarify they should only be used internally. I think pause/resume only make sense internally, For external use, I think we would want to expose `stop` and implement `start`.